tests/kafkatest/services/verifiable_consumer.py

```diff
@@ -173,9 +173,35 @@ def handle_partitions_assigned(self, event, node, logger):
         for topic_partition in event["partitions"]:
             tp = _create_partition_from_dict(topic_partition)
             assignment.append(tp)
-        logger.debug("Partitions %s assigned to %s" % (assignment, node.account.hostname))
+        logger.debug("Incremental: Partitions %s assigned to %s" % (assignment, node.account.hostname))
         self.assignment.extend(assignment)
 
+# This needs to be used for consumer protocol
+class ConsumerProtocolConsumerEventHandler(IncrementalAssignmentConsumerEventHandler):
+    def __init__(self, node, verify_offsets, idx):
+        super().__init__(node, verify_offsets, idx)
+
+    def handle_records_consumed(self, event, logger):
```

Comment:
@dongnuo123 Removing the assertion does not seem right to me because the consumer is not supposed to read records if not joined. I think that we should rather change the on-partition-revoked to not transition to rebalancing. What do you think?

Reply:
Yeah it makes sense to me. Just checked the code and `self.state` here is only used within verifiable_consumer. Then it's ok that the new protocol consumer never gets into `Rebalancing`